src/rules/no-deprecated-functions.ts

```diff
@@ -1,64 +1,26 @@
-import { JSONSchemaForNPMPackageJsonFiles } from '@schemastore/package';
 import {
   AST_NODE_TYPES,
   TSESTree,
 } from '@typescript-eslint/experimental-utils';
+import { JestVersion, detectJestVersion } from './detectJestVersion';
 import { createRule, getNodeName } from './utils';
 
 interface ContextSettings {
   jest?: EslintPluginJestSettings;
 }
 
-export type JestVersion =
-  | 14
-  | 15
-  | 16
-  | 17
-  | 18
-  | 19
-  | 20
-  | 21
-  | 22
-  | 23
-  | 24
-  | 25
-  | 26
-  | 27
-  | number;
-
 interface EslintPluginJestSettings {
-  version: JestVersion;
+  version: JestVersion | string;
 }
 
-let cachedJestVersion: JestVersion | null = null;
-
-/** @internal */
-export const _clearCachedJestVersion = () => (cachedJestVersion = null);
-
-const detectJestVersion = (): JestVersion => {
-  if (cachedJestVersion) {
-    return cachedJestVersion;
+const parseJestVersion = (rawVersion: number | string): JestVersion => {
```

Comment:
Technically this is duplicated in `detectJestVersion`, but I'm fine with that as it's only a few lines and otherwise it calls into question a bit the `JestVersion` & underlying function name which I don't think is worth doing right now.

It'll be easy enough to refactor in the future if we find more uses for this value and/or a use for having minor or patch versions.